### DISCUSSION TITLE
🤌 archived repositories

### DISCUSSION THREAD

**Discussion by landsman** [npm/feedback#613] [Registry]
Hi guys,

one of very annoying thing is legacy packages without maintenace.
I think that this would be somehow flagged in search results, package detail page.

Example:
1. https://www.npmjs.com/search?q=amount
2. so first result is: https://www.npmjs.com/package/amount
3. when you visit repository url: https://github.com/archived-repos/amount (you can see archived warning)

On npm you have to notice just "last publish" of the package.
This waste time of a lot of users.

Suggestions:

A. easy one => user voting to flag this package as archived: let's say 100 votes + old last publish => action
B. harder => cron with simillar rule + check of github api for repository "archived" flag
C. combination of these

What do you think? 🙏

**Comment by ljharb** [comment]
There is absolutely nothing wrong with an “old last publish”; i use and maintain plenty of packages that haven’t updated in years, because they don’t *need* to.

An objective rubric isn’t possible, and a subjective one (voting) would be an easy abuse vector.

**Reply by landsman** [reply]
Yes, sometimes is this ok. In my opinion it's very rare.

But you know cases which I am referring, right?

**Reply by ljharb** [reply]
The number of cases that don’t apply might be larger than the number that do; i think it would be a very bad idea for npm to get involved in judging the quality of packages.

**Reply by landsman** [reply]
let's say we can start with archived github repository, that's clear

**Comment by MylesBorins** [comment]
@landsman there have been some internal discussion about allowing developers to flag that certain packages should be deprecated... a package being deprecated on GitHub but not on npm is definitely a great example of a time where this should perhaps happen. There is definitely a serious risk of incorrectly deprecating something as pointed out by @ljharb . I'll pick this back up in the new year with some more thoughts.

**Reply by landsman** [reply]
Thanks for brainstorming.
